### PR TITLE
Add symbol file override functionality

### PR DESCRIPTION
**Pull request by antonrohr**
This mainly adds/changes functionality in SymbolsDialog. It also
touches the locations where the SymbolsDialog is opened. It also adds
the usage of the overridden file to the symbol loading functionality in
OrbitApp.
Everything is hidden behind the flag --enable_unsafe_symbols

https://screenshot.googleplex.com/3CmdAH8eGRuB9bX
Test: Unit tests and manual testing
http://b/202391472

Apologies for it being a fairly large and rather complicated PR. I decided against adding an extra ItemModel, mainly to save  time/effort, but maybe it is worth doing to decrease complexity. Let me know what you think.

### CODE REVIEW

File: src/ConfigWidgets/SymbolsDialog.cpp [hunk 1]
```diff
@@ -85,12 +149,15 @@ ErrorMessageOr<void> SymbolsDialog::TryAddSymbolPath(const std::filesystem::path
...
-[[nodiscard]] std::vector<std::filesystem::path> SymbolsDialog::GetSymbolPaths() {
+[[nodiscard]] std::vector<std::filesystem::path> SymbolsDialog::GetSymbolPathsFromListWidget() {
   std::vector<std::filesystem::path> result;
-  result.reserve(ui_->listWidget->count());
 
   for (int i = 0; i < ui_->listWidget->count(); ++i) {
-    result.emplace_back(ui_->listWidget->item(i)->text().toStdString());
+    QListWidgetItem* item = ui_->listWidget->item(i);
+    ORBIT_CHECK(item != nullptr);
+    if (dynamic_cast<MappingItem*>(item) != nullptr) continue;
```
Comment (vwbaker) [inline src/ConfigWidgets/SymbolsDialog.cpp:158]: Can you make this if statement if(item->type() != UserType) like you do below? It feels like that would be more consistent.

File: src/ConfigWidgets/SymbolsDialog.cpp [hunk 2]
```diff
@@ -31,22 +32,86 @@ constexpr const char* kOverrideWarningText =
 
 using orbit_client_data::ModuleData;
 using orbit_grpc_protos::ModuleInfo;
+using orbit_object_utils::SymbolsFile;
+
+namespace {
+class MappingItem : public QListWidgetItem {
```
Comment (vwbaker) [inline src/ConfigWidgets/SymbolsDialog.cpp:38]: A comment for the class would be nice. It also might be nice to have the word Override or Force in the class name to make it more clear that this is only used when the user specifically overrides something? But I understand if you don't want to make the name too long, just a thought.

File: src/ConfigWidgets/SymbolsDialog.cpp [hunk 3]
```diff
@@ -31,22 +32,86 @@ constexpr const char* kOverrideWarningText =
...
     : QDialog(parent),
       ui_(std::make_unique<Ui::SymbolsDialog>()),
+      allow_unsafe_symbols_(allow_unsafe_symbols),
       module_(module),
-      persistent_storage_manager_(persistent_storage_manager) {
+      persistent_storage_manager_(persistent_storage_manager),
+      module_symbol_file_mappings_(persistent_storage_manager_->LoadModuleSymbolFileMappings()) {
   ORBIT_CHECK(persistent_storage_manager_ != nullptr);
+
+  // When the symbols dialog is started with an module (from the error) *and* only save symbols are
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply src/ConfigWidgets/SymbolsDialog.cpp:101]: Yeah this whole comment was pretty messy. done

File: src/ConfigWidgets/SymbolsDialog.cpp (hunk 1 above)
Comment (antonrohr) [reply src/ConfigWidgets/SymbolsDialog.cpp:158]: Great idea! done

File: src/ConfigWidgets/SymbolsDialog.cpp (hunk 2 above)
Comment (antonrohr) [reply src/ConfigWidgets/SymbolsDialog.cpp]: I called it `OverrideMappingItem` and added a class description comment. I am not really concerned about naming length, so maybe a even more descriptive name makes sense... maybe `ModuleSymbolFileOverrideItem`.. wdyt?

Comment (vwbaker) [reply src/ConfigWidgets/SymbolsDialog.cpp]: I don't have a strong preference but OverrideMappingItem sounds good to me.

**Review by vwbaker — APPROVED** [review]
I still don't feel like I have a good enough grasp on this code to give the overall architecture question a great review, so another review might be good if you're worried about it but I don't want to block you either. LGTM.

**Review by reichlfl — COMMENTED** [review]
I didn't have time to go through the tests yet

File: src/ConfigWidgets/SymbolsDialog.cpp [hunk 4]
```diff
@@ -31,22 +32,94 @@ constexpr const char* kOverrideWarningText =
...
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply src/ConfigWidgets/SymbolsDialog.cpp]: done

File: src/ConfigWidgets/SymbolsDialog.cpp [hunk 5]
```diff
@@ -165,37 +246,40 @@ void SymbolsDialog::OnAddFileButtonClicked() {
...
+  }
 
-  std::string error = absl::StrFormat(
-      "The build ids of module and symbols file do not match. Module (%s) build id: \"%s\". Symbol "
-      "file (%s) build id: \"%s\".",
-      module.file_path(), module.build_id(), file_path.string(), build_id);
-  return ErrorMessage(error);
+  switch (DisplayOverrideWarning()) {
+    case OverrideWarningResult::kOverride:
+      ORBIT_LOG("override");
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply src/ConfigWidgets/SymbolsDialog.cpp]: You are correct. My mistake

File: src/ConfigWidgets/SymbolsDialog.cpp [hunk 6]
```diff
@@ -165,37 +246,40 @@ void SymbolsDialog::OnAddFileButtonClicked() {
...
+      return AddMapping(module, file_path);
+    case OverrideWarningResult::kCancel:
+      ORBIT_LOG("cancel");
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply src/ConfigWidgets/SymbolsDialog.cpp]: same as above

**Review by reichlfl — APPROVED** [review]
LGTM!

**Pull request merged**